### PR TITLE
server.go restructuring and memory-safe implementation of memory backend

### PR DESCRIPTION
**Pull request by nkashy1**
This is a copy of https://github.com/doc-ai/tensorio-models/pull/9 in which the storage integration with the server which was started in that PR is reverted.

This ensures that `master` always runs some server.

The storage backend integration will be done as a separate PR.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -1,23 +1,32 @@
-package main
+package server
 
 import (
 	"context"
 	"fmt"
+
 	"github.com/doc-ai/tensorio-models/api"
+	"github.com/doc-ai/tensorio-models/storage"
+	log "github.com/sirupsen/logrus"
```
Comment (brianbhsu) [inline server/server.go]: What was the use case for switching from the native logger to logrus?

File: server/server.go [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
-	log.Println(msg.Marker)
-	resp := &api.GetModelsResponse{
+func NewServer(storage storage.RepositoryStorage) api.RepositoryServer {
+	return &server{
+		storage: storage,
+	}
+}
+
+func (srv *server) ListModels(ctx context.Context, req *api.ListModelsRequest) (*api.ListModelsResponse, error) {
+	log.Println(req.Marker)
```
Comment (brianbhsu) [inline server/server.go]: Add more informative log message with method name being called to provide context?

Comment (fkautz) [reply server/server.go]: These will be removed and replaced, they are there for debugging during our scaffolding phase.

File: server/server.go (hunk 1 above)
Comment (fkautz) [reply server/server.go]: logrus provides structured logs compatible with a variety of machine parseable formats that can be ingested by monitoring solutions.

log is just free form text.

File: server/server.go (hunk 2 above)
Comment (fkautz) [reply server/server.go]: Created #12 to track this.

**Pull request merged**